hudi-spark/src/main/java/org/apache/hudi/keygen/SimpleKeyGenerator.java

```diff
@@ -55,21 +51,22 @@ public SimpleKeyGenerator(TypedProperties props, String partitionPathField) {
 
   @Override
   public String getRecordKey(GenericRecord record) {
-    return KeyGenUtils.getRecordKey(record, recordKeyField);
+    return KeyGenUtils.getRecordKey(record, getRecordKeyFields().get(0));
```

Comment:
might be good to assert this out in the constructor itself

Reply:
We wanted to have the same exp as getKey(). we don't throw exception in constructor if record key is not found. we throw only when getKey(GenericRecord record) is called.